swarm/network/stream/common_test.go

```diff
@@ -109,7 +109,7 @@ func newStreamerTester(t *testing.T) (*p2ptest.ProtocolTester, *Registry, *stora
 
 	db := storage.NewDBAPI(localStore)
 	delivery := NewDelivery(to, db)
-	streamer := NewRegistry(addr, delivery, db, state.NewInmemoryStore(), nil)
+	streamer := NewRegistry(addr, delivery, db, state.NewInmemoryStore(), nil, nil)
```

Comment:
i dont understand why we need to know about stream package. 
there should just be `swap.Protocol` wrapper around `p2p.Protocol` and same for Peer no?

Reply:
This is the basic design question and the reason to share this PR and discuss.

In gitter, you said:
> As we had discussed, I would prefer accounting to be an extention of p2p/protocol which overwrites the Send function and the Run function,

I have not been able to understand how to do this. 

`p2p/protocol` interface has no function `Send`, `p2p.Peer` does, and has a `Run` function.
If it is a `p2p/protocol`, how does the Swap protocol get the reference to the message in `Accounting(msg)` from different protocols for accounting?

Thus, the only way I could see this working, is to actually only **replace** the *streamer*'s **embedded** `protocols.Peer` in `stream.Peer` (line 55) by embedding `SwapPeer`.
But this way I  need to pass the `Swap` instance to the registry in order for the streamer peer to get access to it when creating the new streamer `Peer`.

The `p2p.Protocol` stub which is now in `swarm/swap/protocol.go` is only for sending and receiving cheques, thus tracking that.

It is exceedingly important  for me to get his right. I hope I made myself clear on how this is currently implemented.